src/cli_shared/cli/mod.rs

```diff
@@ -186,9 +189,16 @@ impl CliOpts {
         } else {
             cfg.client.enable_rpc = false;
         }
-        if let Some(metrics_address) = self.metrics_address {
-            cfg.client.metrics_address = metrics_address;
+
+        if self.no_metrics {
+            cfg.client.enable_metrics_endpoint = false;
+        } else {
+            cfg.client.enable_metrics_endpoint = true;
+            if let Some(metrics_address) = self.metrics_address {
+                cfg.client.metrics_address = metrics_address;
+            }
         }
```

Comment:
A bit of code-gold if you wish
```suggestion
        cfg.client.enable_metrics_endpoint = !self.no_metrics;
        cfg.client.metrics_address = self.metrics_address.unwrap_or(cfg.client.metrics_address);

```